pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -79,6 +79,9 @@ type KeycloakSpec struct {
 	// Name of the StorageClass for Postgresql Persistent Volume Claim
 	// +optional
 	StorageClassName *string `json:"storageClassName,omitempty"`
+	// Specify PodAntiAffinity settings for Keycloak deployment in Multi AZ
+	// +optional
+	MultiAvailablityZones MultiAvailablityZonesConfig `json:"multiAvailablityZones,omitempty"`
```

Comment:
Something to consider. The approach we took in Infinispan was to allow users to define their own policies by exposing the Affinity spec on our CR, e.g.:

```go
type InfinispanSpec struct {
	...
	Affinity  *corev1.Affinity        `json:"affinity,optional,omitempty"`
}
```

This allows users to define their own Affinity or AntiAffinity policies if they have additional requirements (co locating keycloak/infinispan pods on the same nodes as a performance intensive application for example). In our case, if the user doesn't specify a policy then we provide a default strategy. In our documentation we then have examples of different use-cases and appropriate Affinity strategies.

Here's the Infinispan PR if you're curious https://github.com/infinispan/infinispan-operator/pull/654

Reply:
Myself and @slaskawi had a discussion around this, the first version of PR had the changes this way. Then @slaskawi mentioned/suggested that, we wanted to avoid users configuring low level knobs of the system and wanted to keep the top level keycloak CR as simple as possible. That is the reason we removed the affinity settings from the CR, and kept a boolean enablement feature.